modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_1/SXPBlueprintUpgradeProcess.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.search.experiences.internal.upgrade.v3_1_1;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Gustavo Lima
+ */
+public class SXPBlueprintUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeSXPElement();
+
+		_upgradeSXPBlueprint();
+	}
+
+	private String _fixElementDefinition(String elementDefinition)
+		throws Exception {
+
+		if (Validator.isBlank(elementDefinition)) {
+			return elementDefinition;
+		}
+
+		JSONObject sxpElementJSONObject = JSONFactoryUtil.createJSONObject(
+			elementDefinition);
+
+		if (sxpElementJSONObject == null) {
+			return elementDefinition;
+		}
+
+		JSONObject uiConfigurationJSONObject =
+			sxpElementJSONObject.getJSONObject("uiConfiguration");
+
+		if (uiConfigurationJSONObject == null) {
+			return elementDefinition;
+		}
+
+		JSONArray fieldSetsJSONArray = uiConfigurationJSONObject.getJSONArray(
+			"fieldSets");
+
+		if (fieldSetsJSONArray == null) {
+			return elementDefinition;
+		}
+
+		for (int i = 0; i < fieldSetsJSONArray.length(); i++) {
+			JSONObject fieldSetJSONObject = fieldSetsJSONArray.getJSONObject(i);
+
+			JSONArray fieldsJSONArray = fieldSetJSONObject.getJSONArray(
+				"fields");
+
+			if (fieldsJSONArray == null) {
+				continue;
+			}
+
+			for (int j = 0; j < fieldsJSONArray.length(); j++) {
+				JSONObject fieldJSONObject = fieldsJSONArray.getJSONObject(j);
+
+				JSONArray defaultValueJSONArray = fieldJSONObject.getJSONArray(
+					"defaultValue");
+
+				JSONObject defaultValueJSONObject =
+					fieldJSONObject.getJSONObject("defaultValue");
+
+				if (defaultValueJSONArray != null) {
+					fieldJSONObject.remove("defaultValue");
+					fieldJSONObject.put("fieldMappings", defaultValueJSONArray);
+				}
+				else if (defaultValueJSONObject != null) {
+					fieldJSONObject.put("defaultValue", StringPool.BLANK);
+				}
+			}
+		}
+
+		return sxpElementJSONObject.toString();
+	}
+
+	private String _fixElementInstance(String elementInstances)
```

Comment:
If possible, I'd consider renaming it to something more descriptive. "fixXYZ" doesn't tell much.

Reply:
https://github.com/brianchandotcom/liferay-portal/commit/e871b34f75b1de96155b9ec9b3bf856a468e8f24